rest/src/main/kotlin/com/gitlab/kordlib/rest/Image.kt

```diff
@@ -18,19 +18,42 @@ class Image private constructor(val data: ByteArray, val format: Format) {
         }
 
     companion object {
-        fun raw(data: ByteArray, format: Format): Image {
-            return Image(data, format)
+        fun raw(data: ByteArray, format: Format, size: Size): Image {
+            return Image(data, format, size)
         }
 
         suspend fun fromUrl(client: HttpClient, url: String): Image = with(Dispatchers.IO) {
             val call = client.request<HttpResponse>(url) { method = HttpMethod.Get }
+            val size = Size.fromResolution(call.request.url.parameters["size"])
             val contentType = call.headers["Content-Type"]
-                    ?: error("expected 'Content-Type' header in image request")
+                ?: error("expected 'Content-Type' header in image request")
 
             @Suppress("EXPERIMENTAL_API_USAGE")
             val bytes = call.content.toByteArray()
 
-            Image(bytes, Format.fromContentType(contentType))
+            Image(bytes, Format.fromContentType(contentType), size)
+        }
+    }
+
+    /**
+     * Represents size of the Image as specified by
+     */
+    enum class Size(val resolution: String) {
+        SIZE_16("16"),
+        SIZE_32("32"),
+        SIZE_64("64"),
+        SIZE_128("128"),
+        SIZE_256("256"),
+        SIZE_512("512"),
+        SIZE_1024("1024"),
+        SIZE_2048("2048"),
+        SIZE_4096("4096");
```

Comment:
We generally prefer PascalCase naming for enum and sealed class constants. `Size16` etc would be more in line with the project's conventions.